eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -282,24 +277,37 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
     }
   }
 
-  private def onSuccess(replyTo: ActorRef, result: PaymentSent): Unit = {
-    if (cfg.storeInDb) paymentsDb.updateOutgoingPayment(result)
-    replyTo ! result
-    if (cfg.publishEvent) context.system.eventStream.publish(result)
-    Metrics.SentPaymentDuration
-      .withTag(Tags.MultiPart, if (cfg.id != cfg.parentId) Tags.MultiPartType.Child else Tags.MultiPartType.Disabled)
-      .withTag(Tags.Success, value = true)
-      .record(System.currentTimeMillis - start, TimeUnit.MILLISECONDS)
-  }
-
-  private def onFailure(replyTo: ActorRef, result: PaymentFailed): Unit = {
-    if (cfg.storeInDb) paymentsDb.updateOutgoingPayment(result)
-    replyTo ! result
+  def myStop(request: SendPayment, result: PaymentEvent): State = {
+    if (cfg.storeInDb) {
+      result match {
+        case paymentFailed: PaymentFailed =>
+          paymentsDb.updateOutgoingPayment(paymentFailed)
+        case paymentSent: PaymentSent =>
+          paymentsDb.updateOutgoingPayment(paymentSent)
+        case _ => ()
+      }
+    }
+    request.replyTo ! result
     if (cfg.publishEvent) context.system.eventStream.publish(result)
-    Metrics.SentPaymentDuration
-      .withTag(Tags.MultiPart, if (cfg.id != cfg.parentId) Tags.MultiPartType.Child else Tags.MultiPartType.Disabled)
-      .withTag(Tags.Success, value = false)
-      .record(System.currentTimeMillis - start, TimeUnit.MILLISECONDS)
+    if (cfg.recordMetrics) {
+      val fees = result match {
+        case paymentSent: PaymentSent => paymentSent.feesPaid
+        case _ => 0 msat
+      }
+      val success = result.isInstanceOf[PaymentSent]
+      val now = System.currentTimeMillis
+      val duration = now - start
+      val experimentName = request match {
+        case SendPaymentToNode(_, _, _, _, _, routeParams) => routeParams.experimentName
+        case SendPaymentToRoute(_, _, _, _) => ""
```

Comment:
Why don't you explicitly skip these and only publish events in the `SendPaymentToNode` case?
Since in other places you explicitly try to disable it, it would make more sense, wouldn't it?

Reply:
Done